app/callbacks/data_callbacks.py

```diff
@@ -80,12 +110,26 @@ def login_callback(n_clicks, username, password, user_headers):
                     {"username": username, "password": password},
                     client.headers,
                     dash.no_update,
+                    hide_element_style,
+                    hide_element_style,
+                    hide_element_style,
+                    hide_element_style,
+                    show_spinner_style,
                 )
             except Exception:
                 # This if statement is verified if credentials are invalid
                 form_feedback.append(html.P("Nom d'utilisateur et/ou mot de passe erroné."))
 
-                return dash.no_update, dash.no_update, form_feedback
+                return (
+                    dash.no_update,
+                    dash.no_update,
+                    form_feedback,
+                    input_style_unchanged,
+                    input_style_unchanged,
+                    empty_style_unchanged,
+                    empty_style_unchanged,
+                    hide_element_style,
+                )
```

Comment:
@MateoLostanlen @fe51 @RonanMorgan Another example where `Dash` does not scale well. In a "sane" 😉 web framework:

1. You shouldn't have to repeatedly include placeholders for things you don't want to update: Just pass a map/dict of what you _do_ want to change and the framework should know the rest should be left untouched.
2. You shouldn't have to go back and forth between the parameters in the head of your function and the return to match the order of the declared outputs and with the order of the returns => just return a map each time.